Description: Add spl/arndale-spl.bin rule
Author: Ian Campbell <ijc@debian.org>

Index: u-boot/Makefile
===================================================================
--- u-boot.orig/Makefile
+++ u-boot/Makefile
@@ -1339,6 +1339,9 @@ spl/u-boot-spl-dtb.sfp: spl/u-boot-spl
 spl/boot.bin: spl/u-boot-spl
 	@:
 
+spl/arndale-spl.bin: spl/u-boot-spl
+	@:
+
 tpl/u-boot-tpl.bin: tools prepare
 	$(Q)$(MAKE) obj=tpl -f $(srctree)/scripts/Makefile.spl all
 
