Index: u-boot/include/configs/novena.h
===================================================================
--- u-boot.orig/include/configs/novena.h
+++ u-boot/include/configs/novena.h
@@ -204,7 +204,7 @@
 	"pxefile_addr_r="__stringify(CONFIG_LOADADDR)"\0"               \
 	"scriptaddr="__stringify(CONFIG_LOADADDR)"\0"                   \
 	"ramdisk_addr_r=0x28000000\0"                                   \
-	"fdt_addr_r=0x10000000\0"                                       \
+	"fdt_addr_r=0x11200000\0"                                       \
 	"addcons="							\
 		"setenv bootargs ${bootargs} "				\
 		"console=${consdev},${baudrate}\0"			\
